src/skdh/context/motion_detect.py

```diff
@@ -38,42 +58,301 @@ def __init__(self, filter_cutoff=6, cov_threshold=0.022558169349678834):
+
+    Method implemented as described in:
+    Ji, J., Venderley, J., Zhang, H. et al. Assessing nocturnal scratch with actigraphy in atopic dermatitis
+    patients. npj Digit. Med. 6, 72 (2023). https://doi.org/10.1038/s41746-023-00821-y
+
+    Input requirements:
+
+    1. Accelerometer data is expected to be tri-axial. Orientation does not affect
+    algorithm performance.
+
+    2. Acceleration units are expected to be in G's.
+
+    3. Accelerometer data is expected to be sampled at exactly 20hz.
+
+    4. A minimum of 20 samples (or the equivalent of a single 1-second window) is
+    required for predictions to be made.
+
+    Parameters
+    ----------
+    lp_filter_cutoff : float
+        Low pass filter cutoff. Data filtered prior to CoV calculation.
+    lp_filter_order : int
+        Low pass filter order. Data filtered prior to CoV calculation.
+    hp_filter_cutoff : float
+        High pass filter cutoff. Data filtered prior to CoV calculation.
+    hp_filter_order : int
+        High pass filter order. Data filtered prior to CoV calculation.
+    cov_threshold : float
+        Threshold computed rolling CoV to determine motion present.
+    sd_threshold : float
+        Threshold for max standard deviation to determine motion present.
+
+    """
+
+    def __init__(
+        self,
+        lp_filter_cutoff=3.0,
+        lp_filter_order=6,
+        hp_filter_cutoff=0.25,
+        hp_filter_order=1,
+        cov_threshold=0.41,
+        sd_threshold=0.013,
+    ):
+        super().__init__(
+            lp_filter_cutoff=lp_filter_cutoff,
+            lp_filter_order=lp_filter_order,
+            hp_filter_cutoff=hp_filter_cutoff,
+            hp_filter_order=hp_filter_order,
+            cov_threshold=cov_threshold,
+            sd_threshold=sd_threshold,
+        )
+        self.lp_filter_cutoff = lp_filter_cutoff
+        self.lp_filter_order = lp_filter_order
+        self.hp_filter_cutoff = hp_filter_cutoff
+        self.hp_filter_order = hp_filter_order
+        self.cov_threshold = cov_threshold
+        self.sd_threshold = sd_threshold
+
+    @handle_process_returns(results_to_kwargs=False)
+    def predict(self, time, accel, fs=None, **kwargs):
+        """
+        predict(time, accel, fs)
+
+        Function to detect periods of motion from an accelerometer signal.
+
+        Parameters
+        ----------
+        time : numpy.ndarray
+            (N, ) array of unix timestamps, in seconds.
+        accel : numpy.ndarray
+            (N, 3) array of acceleration, in units of 'g', collected at 20hz.
+        fs : float
+            Sampling rate. Default None. If not provided, will be inferred.
+
+        Returns
+        -------
+        results : dict
+            Results dictionary with motion predictions (1hz boolean; true indicates motion) and associated timestamps.
+
+        """
+        # check input requirements are met
+        time, accel, fs = self._check_input(time, accel, fs)
```

Comment:
if these are the same method, suggest creating a function both will call instead of duplicating. Though see comment in the function as to if this is necessary

Reply:
Not the same method, though similar. Left a version in each without the expensive check for G's